monai/utils/nvtx.py

```diff
@@ -53,12 +55,26 @@ def __init__(
         name: Optional[str] = None,
         methods: Optional[Union[str, Tuple[str, ...]]] = None,
         append_method_name: Optional[bool] = None,
+        recursive: bool = False,
     ) -> None:
         self.name = name
         self.methods = methods
         self.append_method_name = append_method_name
+        self.recursive = recursive
 
     def __call__(self, obj: Any):
+        if self.recursive is True:
+            if isinstance(obj, (list, tuple)):
+                return type(obj)(Range(recursive=True)(t) for t in obj)
+
+            from monai.transforms.compose import Compose
+
+            if isinstance(obj, Compose):
+                obj.transforms = Range(recursive=True)(obj.transforms)
+                return Range(self.name, recursive=False)(obj)
```

Comment:
thanks, looks like this line created an additional `Range` object which is not really needed? 
```suggestion
```

Reply:
It annotates the whole scope of transforms that a compose defines. For instance, we want to annotation "Preprocessing" part that consists many transforms, each of which we also want to be annotated.
```python
    train_preprocess = Compose(
        [
            ToNumpyd(keys="image"),
            RandFlipd(keys="image", prob=0.5),
            RandRotate90d(keys="image", prob=0.5, max_k=3, spatial_axes=(-2, -1)),
            CastToTyped(keys="image", dtype=np.float32),
            RandZoomd(keys="image", prob=0.5, min_zoom=0.9, max_zoom=1.1),
            ScaleIntensityRanged(keys="image", a_min=0.0, a_max=255.0, b_min=-1.0, b_max=1.0),
            ToTensord(keys=("image", "label")),
        ]
    )
    train_preprocess = Range("Preprocessing", recursive=True)(train_preprocess)

```

Reply:
yes, when it reaches line 73, it's already within a call of `Range(recursive=True)(Compose)`, the program can continue the annotating instead of starting a new object and call `Range(recursive=False)(Compose)` again (line 74).